cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -823,15 +914,71 @@ impl Node<Option<cst::VariableDef>> {
                 }
                 op => Err(self.to_ast_err(ToASTErrorKind::InvalidActionScopeOperator(*op))),
             }?;
-            action_constraint
-                .contains_only_action_types()
-                .map_err(|non_action_euids| {
-                    rel_expr
-                        .to_ast_err(parse_errors::InvalidActionType {
-                            euids: non_action_euids,
-                        })
-                        .into()
+            let action_constraint_res = action_constraint.contains_only_action_types();
+
+            action_constraint_res.map_err(|non_action_euids| {
+                rel_expr
+                    .to_ast_err(parse_errors::InvalidActionType {
+                        euids: non_action_euids,
+                    })
+                    .into()
+            })
```

Comment:
this refactor can probably be reverted? the original code was more concise and I'm not sure that giving a name to `action_constraint_res` helps the readability

Reply:
Yup, this made sense in an earlier version of how I wanted to do this and forgot to change it back. Will fix